### PR TITLE
[CARBONDATA-1904][CARBONDATA-1905] Support auto handoff and close streaming

### PR DESCRIPTION
**Pull request by QiangCai**
1. auto handoff streaming segment
2. alter streaming table to normal table
   alter table <tableName> compact 'close_streaming'

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2407/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1183/

**Comment by QiangCai** [comment]
retest this please

**Comment by QiangCai** [comment]
retest sdv please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2423/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2604/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1199/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2614/

**Comment by QiangCai** [comment]
retest this please

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 1]
```diff
@@ -1459,6 +1459,13 @@
    */
   public static final String HANDOFF_SIZE = "carbon.streaming.segment.max.size";
 
+  /**
+   * enable auto handoff streaming segment
+   */
+  public static final String ENABLE_AUTO_HANDOFF = "carbon.enable.auto.handoff";
```
Comment (watermen) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: I think `carbon.streaming.auto.handoff.enabled` is better.

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 2]
```diff
@@ -1459,6 +1459,13 @@
    */
   public static final String HANDOFF_SIZE = "carbon.streaming.segment.max.size";
 
+  /**
+   * enable auto handoff streaming segment
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: Can you describe more how this feature is working, and you can copy the description to document in `/doc` folder

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala [hunk 3]
```diff
@@ -114,6 +114,14 @@ class TestStreamingTableOperation extends QueryTest with BeforeAndAfterAll {
 
     // 14. finish streaming
     createTable(tableName = "stream_table_finish", streaming = true, withBatchLoad = true)
```
Comment (jackylk) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala]: please add a test case to *reopen* the stream property after closing stream

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2432/

File: streaming/src/main/scala/org/apache/spark/sql/execution/streaming/CarbonAppendableStreamSink.scala [hunk 4]
```diff
@@ -80,6 +80,12 @@ class CarbonAppendableStreamSink(
     CarbonProperties.getInstance().getHandoffSize
   )
 
+  // auto handoff
+  private val enableAutoHandoff = hadoopConf.getBoolean(
+    CarbonCommonConstants.ENABLE_AUTO_HANDOFF,
```
Comment (watermen) [inline streaming/src/main/scala/org/apache/spark/sql/execution/streaming/CarbonAppendableStreamSink.scala]: Write the name in doc.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1206/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1217/

File: streaming/src/main/scala/org/apache/spark/sql/execution/streaming/CarbonAppendableStreamSink.scala (hunk 4 above)
Comment (QiangCai) [reply streaming/src/main/scala/org/apache/spark/sql/execution/streaming/CarbonAppendableStreamSink.scala]: ok

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 1 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: ok

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 2 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: ok, I will raise pr to finish streaming doc

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala (hunk 3 above)
Comment (QiangCai) [reply integration/spark2/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOperation.scala]: added

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2482/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1258/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**